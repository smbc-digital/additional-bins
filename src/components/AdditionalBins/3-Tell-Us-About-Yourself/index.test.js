import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { TellUsAboutYourself } from './index'
import renderer from 'react-test-renderer'
import getPageRoute from '../../../helpers/pagehelper'


Enzyme.configure({ adapter: new Adapter() })
describe('TellUsAboutYourself',() => { 


it('should submit the form and go to page 4 when user does not select other on the previous page', () => {
    // Arrange
    const history = { push: jest.fn() }
    const context = {
        address : {
            value: '',
            isValid: false
        },
        firstName : {
            value: '',
            isValid: false
        },
        lastName : {
            value: '',
            isValid: false
        },
        emailAddress : {
            value: '',
            isValid: false
        },
        phoneNumber : {
            value: '',
            isValid: false
        },
        whyMoreSpace : {
            value: 'test',
            isValid: true
        }
    }


    const wrapper = mount(<TellUsAboutYourself context={context} history={history} />)


    // Act
    wrapper.find('form').simulate('submit')


    // Assert
    expect(history.push).toHaveBeenCalledWith(getPageRoute(4))
    })

    it('should submit the form and go to page 6 when user does not select other on the previous page', () => {
        // Arrange
        const history = { push: jest.fn() }
        const context = {
            address : {
                value: '',
                isValid: false
            },
            firstName : {
                value: '',
                isValid: false
            },
            lastName : {
                value: '',
                isValid: false
            },
            emailAddress : {
                value: '',
                isValid: false
            },
            phoneNumber : {
                value: '',
                isValid: false
            },
            whyMoreSpace : {
                value: 'other',
                isValid: true
            }
        }
    
    
        const wrapper = mount(<TellUsAboutYourself context={context} history={history} />)
    
    
        // Act
        wrapper.find('form').simulate('submit')
    
    
        // Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(6))
        })

    it('should find the elements required',() => { 
        // Arrange
        const context = {
            address : {
                value: '',
                isValid: false
            },
            firstName : {
                value: '',
                isValid: false
            },
            lastName : {
                value: '',
                isValid: false
            },
            emailAddress : {
                value: '',
                isValid: false
            },
            phoneNumber : {
                value: '',
                isValid: false
            }
        }
        // Act
        const wrapper = mount(<TellUsAboutYourself context={context} history={history}/>)
        expect(wrapper.find('h1', 'h2', 'TextInputContainer', 'AddressPicker', 'Button', 'Anchor', 'Fragment').exists()).toBe(true)
    })

    it('Should render automatic address', () => {
        const history = { push: jest.fn() }
		const context = {
            address : {
                value: '',
                isValid: false
            },
            firstName : {
                value: '',
                isValid: false
            },
            lastName : {
                value: '',
                isValid: false
            },
            emailAddress : {
                value: '',
                isValid: false
            },
            phoneNumber : {
                value: '',
                isValid: false
            }
        }
    
        const wrapper = mount(<TellUsAboutYourself history={history} context={context}/>)
    
        expect(wrapper.find('#addressLine1').exists()).toBeFalsy()
    })

    describe('snapshot', () => {
    it('renders correctly', () => {
        const context = {
            address : {
                value: 'a',
                isValid: false
            },
            firstName : {
                value: 'b',
                isValid: false
            },
            lastName : {
                value: 'c',
                isValid: false
            },
            emailAddress : {
                value: 'd',
                isValid: false
            },
            phoneNumber : {
                value: 'e',
                isValid: false
            }
        }


        const tree = renderer
            .create(< TellUsAboutYourself context={context} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})

})