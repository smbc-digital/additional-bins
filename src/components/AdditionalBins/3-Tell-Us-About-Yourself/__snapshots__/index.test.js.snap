// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TellUsAboutYourself snapshot renders correctly 1`] = `
Array [
  <form
    onSubmit={[Function]}
  >
    <h1>
      Request an additional black bin
    </h1>
    <h2>
      Tell us about yourself
    </h2>
    <label
      htmlFor="firstName"
    >
      First name
    </label>
    <input
      aria-label="First name"
      aria-live="off"
      className=""
      id="firstName"
      inputMode={undefined}
      maxLength={35}
      min={undefined}
      name="firstName"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value="b"
    />
    <label
      htmlFor="lastName"
    >
      Last name
    </label>
    <input
      aria-label="Last name"
      aria-live="off"
      className=""
      id="lastName"
      inputMode={undefined}
      maxLength={60}
      min={undefined}
      name="lastName"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value="c"
    />
    <h3
      aria-hidden={true}
      className="optional"
    >
      Phone number
    </h3>
    <p
      aria-hidden={true}
      className="optional"
    >
      (optional)
    </p>
    <label
      className="optional sr-only"
      htmlFor="phoneNumber"
    >
      Phone number optional
    </label>
    <input
      aria-label={undefined}
      aria-live={undefined}
      className=""
      id="phoneNumber"
      inputMode={undefined}
      maxLength={60}
      min={undefined}
      name="phoneNumber"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="tel"
      value="e"
    />
    <label
      htmlFor="emailAddress"
    >
      Email address
    </label>
    <input
      aria-label="Email address"
      aria-live="off"
      className=""
      id="emailAddress"
      inputMode={undefined}
      maxLength={60}
      min={undefined}
      name="emailAddress"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="email"
      value="d"
    />
    <label
      htmlFor="postcode"
    >
      Enter your postcode
    </label>
    <input
      aria-label="Enter your postcode"
      aria-live="off"
      className=" address-picker-input"
      id="postcode"
      inputMode={undefined}
      maxLength={undefined}
      min={undefined}
      name="postcode"
      onChange={[Function]}
      onKeyDown={[Function]}
      pattern={undefined}
      type="text"
      value=""
    />
    <button
      className="button-secondary"
      disabled={true}
      onClick={[Function]}
    >
      Find the address
    </button>
    <button
      className="button-primary"
      disabled={true}
      id={undefined}
      onClick={undefined}
      type="submit"
    >
      Next step
    </button>
  </form>,
  <p>
    <a
      href="#"
      onClick={[Function]}
    >
      Previous
    </a>
  </p>,
]
`;
